client/src/components/MoviesPage.jsx

```diff
@@ -3,15 +3,32 @@ import React from 'react';
 import SearchContainer from './SearchContainer';
 import MoviesResults from './MoviesResults';
 
-const MoviesPage = ({ list, filterType, switchFilter, sortType, switchSortType, selectMovie, search }) => (
-    <div>
-        <div>
-            <SearchContainer onSearch={search} filter={filterType} onSwitchFilter={switchFilter} />
-        </div>
-        <div>
-            <MoviesResults list={list} sortType={sortType} switchSortType={switchSortType} selectMovie={selectMovie}/>
-        </div>
-    </div>
-)
+import {getFilterPart, getSortPart} from '../common/queryUtils';
+
+class MoviesPage extends React.Component {
+    onSelectMovie(selectedMovie){
+        if (selectedMovie !== null){
+            this.props.history.push('/film/'+selectedMovie.id);
+        }
+    }
+
+    onSearch(filterValue){
+        this.props.search(filterValue);
```

Comment:
_trivial_
It seems, that this method does too much.

On this training project, I don't insist on refactoring this, but on a prod project you can consider please consider an alternative e.g.

IF you do here:
```javascript
  this.props.search(filterValue);
```
I assume it will result in creating an action, THEN you can define a separate saga that will listen to that action and perform route change.

Another way might be leaving this:
```javascript
  this.props.history.push('/search/'+getFilterPart(this.props.filterType, filterValue)+getSortPart(this.props.sortType));
```
and THEN run some logic related to searching but within that component that will be resolved by React router after `hitstory.push`.

On areal projects, such kinds of methods tend to grow in terms of responsibilities that they might have.

For instance, later you might have a requirement to attach some sort of analytics to be executed each time user fulfil a search for movies etc and after all, it may become unmanageable at all.